src/main/java/core/basesyntax/Storage.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+
+    private Map<String, Integer> fruits = new HashMap<>();
+
+    public void setFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, quantity);
+    }
+
+    public void addFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public void reduceFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) - quantity);
```

Comment:
Handling Purchase operation: check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.

Reply:
changes have been made